contracts/LibSubmarineSimple.sol

```diff
@@ -0,0 +1,184 @@
+pragma solidity ^0.4.24;
+
+import "./SafeMath.sol";
+import "./proveth/ProvethVerifier.sol";
+
+contract LibSubmarineSimple is ProvethVerifier {
+
+    using SafeMath for uint256;
+
+    ////////////
+    // Events //
+    ////////////
+
+    event Unlocked(
+        bytes32 indexed _commitId,
+        uint256 _commitValue
+    );
+    event Revealed(
+        bytes32 indexed _commitId,
+        uint256 _commitValue,
+        bytes32 _witness,
+        bytes32 _commitBlockHash,
+        address submarineAddr
+    );
+
+    /////////////
+    // Storage //
+    /////////////
+
+    uint8 public vee = 27; // the ECDSA v parameter: 27 allows us to be broadcast on any network (i.e. mainnet, ropsten, rinkeby etc.)
+    uint32 public commitPeriodLength; // How many blocks must a submarine be committed for before being revealed
+
+    mapping(bytes32 => CommitData) public commitData; // stored "session" state information
+
+    // A submarine send is considered "finished" when the amount revealed and unlocked are both greater than zero, and the amount for the unlock is greater than or equal to the reveal amount.
+    struct CommitData {
+        uint128 amountRevealed; // amount the reveal transaction revealed would be sent in wei. When greater than zero, the submarine has been revealed.
+        uint128 amountUnlocked; // amount the unlock transaction recieved in wei. When greater than zero, the submarine has been unlocked; however the submarine may not be finished, until the unlock amount is GREATER than the promised revealed amount.
+    }
+    
+    // Response from proveth Merkle Patricia Proof Struct
+    struct ProvenTransaction{
```

Comment:
Isn't the original struct available since we inherit from ProvethVerifier?

Reply:
correct fixed in later commit